sample-app/src/main/java/com/joinforage/android/example/ui/pos/POSComposeApp.kt

```diff
@@ -507,10 +512,25 @@ fun POSComposeApp(
                     merchant = uiState.merchant,
                     terminalId = k9SDK.terminalId,
                     paymentMethod = uiState.tokenizedPaymentMethod,
-                    txType = uiState.refundPaymentResponse?.let { it1 -> TxType.forReceipt(it1.receipt) },
+                    txType = uiState.refundPaymentResponse?.let { it1 ->
+                        it1.receipt?.let { it2 ->
+                            TxType.forReceipt(
+                                it2
+                            )
+                        }
+                    },
                     refundResponse = uiState.refundPaymentResponse!!,
+                    fetchedPayment = uiState.capturePaymentResponse,
+                    onRefundRefClicked = { paymentRef, refundRef -> viewModel.fetchRefund(paymentRef, refundRef) },
```

Comment:
I'm still a little bit bitter that the `Refund` endpoint requires the `paymentRef` in the URL 🙄

Reply:
Yeahhh maybe not ideal